spark/src/main/scala/org/apache/iceberg/spark/SparkTableUtil.scala

```diff
@@ -303,29 +370,38 @@ object SparkTableUtil {
     }
   }
 
-  private def buildManifest(conf: SerializableConfiguration,
-      sparkDataFiles: Seq[SparkDataFile],
-      partitionSpec: PartitionSpec,
-      basePath: String): Iterator[Manifest] = {
-    if (sparkDataFiles.isEmpty) {
-      Seq.empty.iterator
-    } else {
+  private def toSparkPartition(partition: CatalogTablePartition, table: CatalogTable): SparkPartition = {
+    val uri = partition.storage.locationUri.map(String.valueOf(_))
+    require(uri.nonEmpty, "Partition URI should be defined")
+
+    val format = partition.storage.serde.orElse(table.provider)
+    require(format.nonEmpty, "Partition format should be defined")
+
+    SparkPartition(partition.spec, uri.get, format.get)
+  }
+
+  private def buildManifest(
```

Comment:
It is better to have it as an iterator to iterator transformation to avoid materializing data for the partition.